File: ems-backend/src/main/java/com/rizwan/ems_backend/service/DesignationService.java
package com.rizwan.ems_backend.service;

import java.util.List;

import com.rizwan.ems_backend.entity.Designation;

public interface DesignationService {

    Designation saveDesignation(Designation designation);

    List<Designation> getAllDesignations();

    Designation getDesignationById(Long id);

    Designation updateDesignation(Long id, Designation designation);

    void deleteDesignation(Long id);
}

File: ems-frontend/src/services/DesignationService.js
import axios from "axios";

const REST_API_BASE_URL = "http://localhost:8080/api/designations";

class DesignationService {

    getAllDesignations() {
        return axios.get(REST_API_BASE_URL);
    }

    getDesignationById(id) {
        return axios.get(REST_API_BASE_URL + "/" + id);
    }

    createDesignation(designation) {
        return axios.post(REST_API_BASE_URL, designation);
    }

    updateDesignation(id, designation) {
        return axios.put(REST_API_BASE_URL + "/" + id, designation);
    }

    deleteDesignation(id) {
        return axios.delete(REST_API_BASE_URL + "/" + id);
    }

}

export default new DesignationService();